elastic/kibana
# [ResponseOps][Window Maintenance] Add window maintenance to alerting plugin public folder and add create form

Resolves https://github.com/elastic/kibana/issues/152272

## Summary

- This PR adds maintenance windows to the alerting plugin and also adds the create form. 
- The Maintenance Windows feature is hidden by a feature flag by default.
- The create form calls the create api, but it's expected to fail because the api hasn't been merged. 

<img width="1709" alt="Screen Shot 2023-04-03 at 2 56 45 PM" src="https://user-images.githubusercontent.com/109488926/229601948-13a46847-bebc-4674-9446-55163a6a6589.png">

<img width="1706" alt="Screen Shot 2023-04-03 at 2 56 56 PM" src="https://user-images.githubusercontent.com/109488926/229601997-03b232cf-0d50-4047-a92b-79ef0c1e9568.png">

<img width="1706" alt="Screen Shot 2023-04-03 at 2 57 12 PM" src="https://user-images.githubusercontent.com/109488926/229602015-7e0110e6-dd3d-41c4-8601-64943fe01323.png">


### Checklist



### To verify
- Enable maintenance windows by adding `ENABLE_MAINTENANCE_WINDOWS = true` to `x-pack/plugins/alerting/common/index.ts`
- Go to http://localhost:5601/app/management/insightsAndAlerting/maintenanceWindows or select the Maintenance Windows link on the side nav
- Click the `Create a maintenance window` button and verify it takes you to the create from
- Verify the bread crumbs and the return link work correctly
- Once on the create form, try to submit without entering any values and verify that the form doesn't submit and errors are shown on the `name` and `duration` fields 
- Submit a maintenance window that is not recurring. You'll see an error popup but verify that the request body is correct and matches what you've selected in the form.
- Create a maintenance window that is recurring and try choosing the different options on the recurring form. Verify that the summary at the bottom of the form accurately reflects what you have selected.
- Submit a maintenance window that is recurring. Again, you'll see an error popup but verify that the request body is correct and matches what you've selected in the form.
